pkg/host/state_machine.go

```diff
@@ -0,0 +1,133 @@
+package host
+
+import (
+	"context"
+	"sync/atomic"
+)
+
+// HostStatus is the status of the host
+type HostStatus string
+
+// Equals returns true if given SNRHostStatus matches this one
+func (s HostStatus) Equals(other HostStatus) bool {
+	return s == other
+}
+
+// IsNotIdle returns true if the SNRHostStatus != Status_IDLE
+func (s HostStatus) IsIdle() bool {
+	return s != Status_IDLE
+}
+
+// IsStandby returns true if the SNRHostStatus == Status_STANDBY
+func (s HostStatus) IsStandby() bool {
+	return s == Status_STANDBY
+}
+
+// IsReady returns true if the SNRHostStatus == Status_READY
+func (s HostStatus) IsReady() bool {
+	return s == Status_READY
+}
+
+// IsConnecting returns true if the SNRHostStatus == Status_CONNECTING
+func (s HostStatus) IsConnecting() bool {
+	return s == Status_CONNECTING
+}
+
+// IsFail returns true if the SNRHostStatus == Status_FAIL
+func (s HostStatus) IsFail() bool {
+	return s == Status_FAIL
+}
+
+// IsClosed returns true if the SNRHostStatus == Status_CLOSED
+func (s HostStatus) IsClosed() bool {
+	return s == Status_CLOSED
+}
+
+func (sm *SFSM) GetCurrent() string {
```

Comment:
nit: This doesn't need to be a pointer.

Reply:
fair, the idea was to save a bit of memory not copying the value every time. but im not totally sure if the type interface creates a struct wrapper or the compiler swaps it out for the primitive type. either way i think it does save us a bit of memory using a pointer here.